src/core/PrettyPrinter.js

```diff
@@ -30,6 +30,8 @@ getJasmineRequireObj().pp = function(j$) {
         this.emitScalar('HTMLNode');
       } else if (value instanceof Date) {
         this.emitScalar('Date(' + value + ')');
+      } else if (value && value.toString && typeof value === 'object' && !(value instanceof Array) && value.toString !== Object.prototype.toString) {
```

Comment:
All but the first (as you rightly point out)

* The `value.toString` check ensures we have a definition for `toString` (if we have a null prototype we won't, it may have explicitly been set to null etc)
* We want to limit this to objects that aren't arrays
* And then the final check ensures we have an actual override of the `toString` method